### PR TITLE
Memory changes for sycl

### PR DESCRIPTION
**Pull request by roigcarlo**
Some memory allocation changes to allow the sycl branch to access data from containers and spans data views.
Mainly changes:
- pressure_operator.hpp
- run.cpp

Also replacing `std::vector<bool>` to `std::vector<int>` because no one knows what kind of specialization details that class has.

### CODE REVIEW

File: cpp/includes/deflation_pressure_preconditioner.hpp [hunk 1]
```diff
@@ -88,7 +88,7 @@ class DeflationPressurePreconditioner : public PressurePreconditioner<TDim>
         VectorType aux_out(mProblemSize);
         for (unsigned int j = 0; j < TDim + 1; ++j) {
             for (unsigned int i = 0; i < mProblemSize; ++i) {
-                aux_in(i) = mRigidModes
+                aux_in(i) = mRigidModes;
```
Comment (rubenzorrilla) [inline cpp/includes/deflation_pressure_preconditioner.hpp:91]: :open_mouth: how the hell was this working?

Comment (roigcarlo) [reply cpp/includes/deflation_pressure_preconditioner.hpp:91]: :man_shrugging: it did not work on my computer, that why I changed it XD

**Review by rubenzorrilla — APPROVED** [review]

File: cpp/includes/deflation_pressure_preconditioner.hpp (hunk 1 above)
Comment (rubenzorrilla) [reply cpp/includes/deflation_pressure_preconditioner.hpp:91]: It was in mine's :smile:

Comment (loumalouomega) [reply cpp/includes/deflation_pressure_preconditioner.hpp:91]: Lo hizo un mago

Comment (rubenzorrilla) [reply cpp/includes/deflation_pressure_preconditioner.hpp:91]: > Lo hizo un mago

Javi Potter García.

**Pull request merged**